### ISSUE TITLE
missing cookbook error

### ISSUE THREAD

**Issue by grizo** [chef-cookbooks/compat_resource#25]
I believe its related to compat_resource (maybe chef gem or audit) but getting the following error when trying to perform `chef-shell -z`
```
epic fail!

/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/cookbook/cookbook_collection.rb:38:in `block in initialize': Cookbook  not found. If you're loading  from another cookbook, make sure you configure the dependency in your metadata (Chef::Exceptions::CookbookNotFound)
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/ohai-8.7.0/lib/ohai/mash.rb:77:in `yield'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/ohai-8.7.0/lib/ohai/mash.rb:77:in `default'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/ohai-8.7.0/lib/ohai/mash.rb:77:in `default'
	from /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/recipe.rb:10:in `[]'
	from /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/recipe.rb:10:in `new'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/shell/shell_session.rb:61:in `block in reset!'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/shell/shell_session.rb:101:in `loading'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/shell/shell_session.rb:54:in `reset!'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/shell.rb:128:in `session'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/shell.rb:137:in `init'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/shell.rb:65:in `start'
	from /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/bin/chef-shell:37:in `<top (required)>'
	from /usr/bin/chef-shell:54:in `load'
	from /usr/bin/chef-shell:54:in `<main>'
```

chef-client executes fine however, even see the following at the end of the run:
```
Deprecated features used!
  chef_gem[chef-handler-mail] chef_gem compile_time installation is deprecated at 1 location:
    - /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef.rb:20:in `log_deprecation'
  chef_gem[chef-handler-mail] Please set `compile_time false` on the resource to use the new behavior. at 1 location:
    - /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef.rb:20:in `log_deprecation'
  chef_gem[chef-handler-mail] or set `compile_time true` on the resource if compile_time behavior is required. at 1 location:
    - /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef.rb:20:in `log_deprecation'
```

currently using compat_resource 12.5.16 and chef client 12.5.1

**Comment by jkeiser** [comment]
Hmm, it's acting a little like chef-shell is sending a `nil` cookbook_name. Can probably fix by adding `cookbook_name &&` to the beginning of this if: https://github.com/chef-cookbooks/compat_resource/blob/master/files/lib/chef_compat/monkeypatches/chef/recipe.rb#L8

Are you able to give that a whirl? I'm not 100% sure how to repro yet (gotta figure out how to get chef-shell going).

**Comment by PhilGrayson** [comment]
You can target a previous version of this cookbook until a fix has been pushed and tagged.

For example in Berkshelf;

`cookbook 'compat_resource', git: 'https://github.com/chef-cookbooks/compat_resource.git', tag: 'v12.5.14'`

**Comment by grizo** [comment]
@jkeiser Tried and no luck, even the latest version.

@PhilGrayson Yeah, I ended up pinning it down just now. Wasn't a blocker but its nice to run chef-shell for some tests at times. Thanks!

**Comment by grizo** [comment]
looks like 12.7.1 version of the cookbook helped this out for chef-client version 12.5.x through 12.7.x

**Comment by itmustbejj** [comment]
This seems to be fixed for me after upgrading to chefdk 0.11.2.
